group/views.py

```diff
@@ -115,56 +157,95 @@ def detail(request, group_id):
         elif contest.end_time < now:
             ended_contest_list.append(contest)
 
+    paginator = Paginator(student_list, 15)  # Show 25 users per page
+    page = request.GET.get('page')
+
+    try:
+        student_list = paginator.page(page)
+    except PageNotAnInteger:
+        # If page is not an integer, deliver first page.
+        student_list = paginator.page(1)
+    except EmptyPage:
+        # If page is out of range (e.g. 9999), deliver last page of results.
+        student_list = paginator.page(paginator.num_pages)
+
     return render_index(
         request, 'group/groupDetail.html', {
-            'rc_list': running_contest_list[0:show_number],
-            'ec_list': ended_contest_list[0:show_number],
-            'an_list': annowence_list,
+            'running_contest_list': running_contest_list[0:show_number],
+            'ended_contest_list': ended_contest_list[0:show_number],
+            'announce_list': annowence_list,
             'coowner_list': coowner_list,
             'owner': owner,
-            's_list': student_list,
-            'group_name': group.gname,
+            'student_list': student_list,
+            'group_name': group.gname, 
             'group_description': group.description,
             'group_id': group.id,
             'user_is_owner': user_is_owner,
+            'user_is_coowner': user_is_coowner,
+            'form': form,
         })
 
 
 def list(request):
```

Comment:
In the future, I think list need to be 2 functions: ```all_group_list``` and ```my_group_list``` to enhance performance of this page.
And if yes, we can create API to get difference groups of different purposes